packages/intrepid2/src/Projection/Intrepid2_ProjectionTools.hpp

```diff
@@ -493,6 +505,90 @@ class ProjectionTools {
+
+    const ViewType4 elemDof_; // rank-1 view having dimension n, containing the basis numbering
+    ordinal_type n_, m_;      // basis cardinality and dimension of the constraint of the KKT system
+
+    /** \brief  Functor constructor
+
+        \code
+          C  - num. cells
+          P  - num. evaluation points
+        \endcode
+
+
+        \param  basisCoeffs           [out]     - rank-2 view (C,F) containing the basis coefficients
+        \param  elemMat               [in/out]  - rank-3 view (C,P,P) containing the element matrix of size
+                                                  numCells x (n+m)x(n+m) on each cell
+                                                  it will be overwritten.
+        \param  elemRhs               [in/out]  - rank-2 view (C,P) containing the element rhs on each cell
+                                                  of size numCells x (n+m)
+                                                  it will contain the solution of the system on output
+        \param  tau                   [out]     - rank-2 view (C,P) used to store the QR factorization
+                                                  size: numCells x (n+m)
+        \param  w                     [out]     - rank-2 view (C,P) used has a workspace
+                                                  Layout Right, size: numCells x (n+m)
+        \param  elemDof               [in]      - rank-1 view having dimension n, containing the basis numbering
+        \param  n                     [in]      - ordinal_type, basis cardinality
+        \param  m                     [in]      - ordinal_type, dimension of the constraint of the KKT system
+     */
+    SolveSystem (ViewType1 basisCoeffs, ViewType2 elemMat, ViewType2 elemRhs, ViewType2 tau,
+        ViewType3 w,const  ViewType4 elemDof, ordinal_type n, ordinal_type m=0) :
+          basisCoeffs_(basisCoeffs), elemMat_(elemMat), elemRhs_(elemRhs),
+          tau_(tau), w_(w), elemDof_(elemDof), n_(n), m_(m){};
+
+
+    void
+    KOKKOS_INLINE_FUNCTION
+    operator()(const ordinal_type ic) const {
+      auto A = Kokkos::subview(elemMat_, ic, Kokkos::ALL(), Kokkos::ALL());
+      auto b = Kokkos::subview(elemRhs_, ic, Kokkos::ALL());
+      auto tau = Kokkos::subview(tau_, ic, Kokkos::ALL());
+      auto w = Kokkos::subview(w_, ic, Kokkos::ALL());
+
+      for(ordinal_type i=n_; i<n_+m_; ++i)
+        for(ordinal_type j=0; j<n_; ++j)
+          A(i,j) = A(j,i);
+
+      //computing QR of A. QR is saved in A and tau
+      KokkosBatched::SerialQR_Internal::invoke(A.extent(0), A.extent(1),
+          A.data(), A.stride_0(), A.stride_1(), tau.data(), tau.stride_0(), w.data());
+
+      //b'*Q -> b
+      KokkosBatched::SerialApplyQ_RightNoTransForwardInternal::invoke(
+          1, A.extent(0), A.extent(1),
+          A.data(),  A.stride_0(), A.stride_1(),
+          tau.data(), tau.stride_0(),
+          b.data(),  b.stride_1(), b.stride_0(),
```

Comment:
Your subview of b is rank 1 and you use strides as like treating b as a matrix. This should be wrong. Once you access b.stride_1(), kokkos should consider this as an error as it attempts to access the second stride for a rank 1 view. The argument should be ``b.data(), 1, b.stride_0(),``. Here 1 is a dummy value and any number is fine.

Reply:
@kyungjoo-kim thanks, I'll fix that.

Reply:
@kyungjoo-kim b.t.w. I guess it doesn't throw an error because it's a DynRankView. anyway, I'll put 1.

Reply:
Yes, it does not really matter. Those strides are always initialized as 1. Someday it is possible that kokkos add more strict rules. So, it is safer not to abuse the interface now.